api/src/pcapi/core/finance/api.py

```diff
@@ -2689,3 +2690,83 @@ def recredit_underage_users() -> None:
     logger.info("Recredited %s users successfully", total_users_recredited)
     if failed_users:
         logger.error("Failed to recredit %s users: %s", len(failed_users), failed_users)
+
+
+def update_bank_account_venues_links(
+    user: users_models.User,
+    offerer: offerers_models.Offerer,
+    bank_account: models.BankAccount,
+    venues_ids: set[int],
+) -> None:
+    managed_venues_ids = {venue.id for venue in offerer.managedVenues}
+    current_links_by_venue_id = {link.venueId: link for link in bank_account.venueLinks}
+
+    venues_already_linked = set(current_links_by_venue_id.keys())
+    venues_links_to_create = venues_ids.difference(venues_already_linked)
+    venues_links_to_deprecate = venues_already_linked.difference(venues_ids)
+
+    new_links = []
+
+    with transaction():
+        link_bulk_update_mapping = []
+        action_history_bulk_insert_mapping = []
+
+        for venue_id in venues_links_to_deprecate:
+            link = current_links_by_venue_id[venue_id]
+
+            link_bulk_update_mapping.append(
+                {
+                    "id": link.id,
+                    "timespan": db_utils.make_timerange(link.timespan.lower, datetime.datetime.utcnow()),
+                }
+            )
+            action_history_bulk_insert_mapping.append(
+                {
+                    "actionType": history_models.ActionType.LINK_VENUE_BANK_ACCOUNT_DEPRECATED,
+                    "userId": user.id,
```

Comment:
La colonne `userId` indique l'utilisateur auquel l'action s'applique (exemples : celui qui est suspendu, celui qui est rattaché, etc.). Remplir cette colonne fait apparaître l'action dans l'historique de cet utilisateur, ce qui n'est probablement pas souhaité.
Je pense que tu voulais plutôt indiquer l'auteur de l'action :
```suggestion
                    "authorUserId": user.id,
```

Reply:
En effet !